index.js

```diff
@@ -7,12 +18,95 @@ function ask(questionText) {
   });
 }
 
-start();
+// starts the pgm - 
+start();   
 
+// function starting the pgm
 async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
+  startGameSetup();
+  // hard coding to min & max to begin with 
+    let min=1;    
+    let max=100;  
+    range = ((max - min)+1);  
+
+    console.log("Please think of a number between " + min + " and " + max + " (inclusive), in this " + range + " number range.");
+    let secretNumber = await ask("What is your secret number?\nThis Javascript (JS) program (pgm) won't peek...\n");
+    console.log('You entered: ' + secretNumber);  
+  
+    let pgmTries = 1;
+    let cpuGuessNum = getRandomIntInclusive(min, max);
+
+    console.log('(This JS pgm randomly guesses only once) Is it... ' + cpuGuessNum);   
+    let userRespYorN = await ask('? ');
+  // in case user used lower case - 
+    let userRespYorNUC = userRespYorN.toUpperCase();  
+    
+  // if correctAnswer();  
+    if (userRespYorNUC === 'Y') {
+    console.log('You stated that this was the correct number, Yea!');
+  } else if (userRespYorNUC == 'N') {
+    console.log('You stated it was not correct, or ' + userRespYorNUC + ' , Sorry.');
+  } else {
+    cheatMsg ();
+  }
+
+    console.log('The pgm has tried: ' + pgmTries + ' times.');
+    let maxTries = (Math.round((Math.log2((max)+1))));
+    console.log('The maximum number of tries this program should theorectially only take is: ' + maxTries);
+
+    //First Loop -
+    // If it was a Y, then pgm needs to stop, but if it was a N, then continue - 
+    if (userRespYorNUC === 'N') { 
+      continueTrying = (true);}    
+    let userRespHorL = await ask("Is it higher (H) or lower (L)? ");
+    
+    // in case the user used lower case - 
+    let userRespHorLUC = userRespHorL.toUpperCase();  
+    
+    // goes to High Loop -  
+     if (userRespHorLUC == 'H') {
+    (gotoHighLoop = (true)); 
+       
+       // needs to go to Low loop - 
+   } else if (userRespHorLUC == 'L') {
+     (gotoLowLoop = (true));
+
+     // else needs to go to end of pgm after - 
+   } else {
+     CheatMsg();
```

Comment:
you capitalized the c in this function call to cheatMsg(). This will throw an error because CheatMsg is different from cheatMsg

Reply:
In addition, this function crashes upon entering a response other than yes, no, higher or lower, stating that userRespYorN isn't defined. This is because you are referencing a variable from one function that is locally available in another.

Reply:
fixed by having different error msgs as well as referencing variables within the error stmt itself